backend/src/common/types/repository.type.ts

```diff
@@ -1,6 +1,7 @@
 type Repository<T = unknown> = {
     find(): Promise<T>;
     findAll(): Promise<T[]>;
+    findByEmail(email: string): Promise<T>;
```

Comment:
we will have this method only in user repo so there is no need to put this in type here